### PR TITLE
doc: minor docs updates for sync

### PR DESCRIPTION
**Pull request by darahayes**
## Purpose

The main purpose is to add the appropriate changes for sync that are being introduced in the upcoming aerogear-js-sdk release. There is a [release candidate](https://github.com/aerogear/aerogear-js-sdk/releases/tag/2.8.0-rc1) out right now and the full release coming soon.

The main changes in there are actually coming from a new release of `Offix`. The docs changes you see in here mostly correspond with what can be seen in the [Offix Release Notes](https://github.com/aerogear/offix/releases/tag/0.9.0)

Type:
- [x] Technical change
- [ ] Style edit

Check that netlify build displays your changes correctly

PS.: Add images and/or .gifs to illustrate what was changed if this pull request modifies the appearance/output of something presented to the users.

### CODE REVIEW

File: modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc [hunk 1]
```diff
@@ -59,36 +59,9 @@ setupClient();
...
-  });
-----
-
-NOTE: In addition to watching individual mutations, you can add a global offline listener when creating a client as described in xref:sync-client-offline-queue-listener[].
-
 = Performing mutations while offline
 
-The @aerogear/voyager-client module provides an `offlineMutate` method which extends Apollo's mutate function with some extra functionality. 
+The @aerogear/voyager-client module provides an `offlineMutate` method which extends Apollo's `mutate` function with some extra functionality. 
 This includes automatically adding some fields to each operation's context.
```
[the remark this answers is not in the document]
Comment (darahayes) [reply modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc]: Good question. It's a mostly internal class that the Apollo Client uses to represent both Queries and Mutations. I kinda took the approach of only modifying the things in the docs that I really needed to modify in order to be accurate. However if you ask me I think that this whole description of `offlineMutate` could be rewritten and does not need to introduce the concept of an operation at all. I will update that shortly!

[the remark this answers is not in the document]
Comment (darahayes) [reply modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc]: I've added a totally new description of the `offlineMutate` method. Let me know if that makes more sense!

File: modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc [hunk 2]
```diff
@@ -116,6 +94,30 @@ Note: The `offlineMutate` method accepts the same parameters as `mutate` with so
...
 
+= Handling mutations while offline
+
+If a mutation occurs while the device is offline, the `client.offlineMutate` function:
+
+- returns immediately
+- returns a promise with an error
+
+You can check the _error_ object to isolate errors relating to offline state.
+Invoking the `watchOfflineChange()` method on an _error_ object watches for when an offline change is synced with the server, and sends a notification when triggered.
```
Comment (briandooley) [inline modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc]:
```suggestion
Invoking the `watchOfflineChange()` method on an error object watches for when an offline change is synced with the server, and sends a notification when triggered.
```

File: modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc [hunk 3]
```diff
@@ -186,11 +181,54 @@ To handle all notifications about offline related events, use the *offlineQueueL
 
 The following events are emitted:
 
-* `onOperationEnqueued` - Called when new operation is being added to offline queue
+* `onOperationEnqueued` - Called when a new operation is added to offline queue
 * `onOperationSuccess` - Called when back online and operation succeeds
 * `onOperationFailure` - Called when back online and operation fails with GraphQL error
+* `onOperationRequeued` - Called when an operation has placed back on the offline queue from storage. This typically happens on application restarts.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/data-sync/js-client-offline.adoc]:
```suggestion
* `onOperationRequeued` - Called when an operation has been placed back on the offline queue from storage. This typically happens on application restarts.
```

**Pull request merged**